### PR TITLE
Merge itemModel.ts and App.ts into Main

### PR DESCRIPTION
**Pull request by jen-chieh**
1. added methods GetItembyID and CreateItem

### CODE REVIEW

File: App.ts [hunk 1]
```diff
@@ -49,15 +49,24 @@ class App {
...
     /* Item Routes */
     router.get('/api/items/', (req, res) => {
       console.log('Query All items');
       this.Items.retrieveAllItems(res);
     });
+    router.get('/api/getItem/:itemID', (req, res) => {
+      console.log('Query item with itemID');
+      this.Items.getItem(req,res);
+    });
+    router.post('/api/createItem', (req, res) => {
```
Comment (Nirmalya24) [inline App.ts:66]: A test to see if the req.body was passed in correctly would be useful

File: model/ItemModel.ts [hunk 2]
```diff
@@ -46,6 +55,57 @@ class ItemModel {
             response.json(numberOfLists) ;
         });
     }
+    /**
+     * Retrieve a single item by ID
+     *@param response - response object
+     *@param filter - filter object
+     *@returns - JSON Object
+     */
+   public retrieveItemByID(response: any, filter: Object) {
```
Comment (Nirmalya24) [inline model/ItemModel.ts:64]: This method should be deleted

File: model/ItemModel.ts [hunk 3]
```diff
@@ -46,6 +55,57 @@ class ItemModel {
...
+     *@param request - request object
+     *@param response - response object
+     */
+
+    public getItem(request:any,response:any): any {
+        const params= request.params;
+        console.log(params);
+        var query = this.model.findOne(params);
+        query.exec( (err, itemArray) => {
+            response.json(itemArray) ;
```
Comment (Nirmalya24) [inline model/ItemModel.ts:85]: Add an error check in case the query fails to execute

File: model/RestaurantModel.ts [hunk 4]
```diff
@@ -45,5 +45,9 @@ class RestaurantModel {
         });
     }
 
+    public getRestaurantByID(response: any, filter: Object): any {
+        // TODO: Implement this
```
Comment (Nirmalya24) [inline model/RestaurantModel.ts:49]: This needs to be filled in

**Review by Nirmalya24 — CHANGES_REQUESTED** [review]
Left review comments before this can be merged into main

File: model/ItemModel.ts (hunk 2 above)
Comment (jen-chieh) [reply model/ItemModel.ts:65]: done

File: model/ItemModel.ts (hunk 3 above)
Comment (jen-chieh) [reply model/ItemModel.ts:88]: done

File: model/RestaurantModel.ts (hunk 4 above)
Comment (jen-chieh) [reply model/RestaurantModel.ts:50]: done

**Comment by jen-chieh** [comment]
1.added error handling in ItemModel.ts
2.fulled method getRestaurantByID in RestaurantModel.ts
3. fixed router in App.ts

**Review by Nirmalya24 — APPROVED** [review]

**Pull request merged**